docs: server-only/client-only in Next.js (#80408)
vercel/next.js: docs/01-app/01-getting-started/08-server-and-client-components.mdx

Now, if you try to import the module into a Client Component, there will be a build-time error.

The corresponding [`client-only` package](https://www.npmjs.com/package/client-only) can be used to mark modules that contain client-only logic like code that accesses the `window` object.

> In Next.js, installing `server-only` or `client-only` is **optional**. However, if your linting rules flag extraneous dependencies, you may install them to avoid issues.

Reviewer: Unused? 

```suggestion
In Next.js, installing `server-only` or `client-only` is **optional**. However, if your linting rules flag unused dependencies, you may install them to avoid warnings.
```

Author: Not really, this was more about covering base for importing a dependency that's not defined in your `package.json`, for example:

- https://biomejs.dev/linter/rules/no-undeclared-dependencies/

> In Next.js, installing `server-only`, or `client-only` is **optional**. Linting rules that prevent usage of extraneous dependencies, not declared in `package.json` may raise an issue if `server-only` or `client-only`, are not installed.

Author: ```suggestion
In Next.js, installing `server-only` or `client-only` is **optional**. However, if your linting rules flag extraneous dependencies, you may install them to avoid issues.
```

```bash package="npm"
npm install server-only
```

```bash package="yarn"
yarn add server-only
```

```bash package="pnpm"
pnpm add server-only
```

```bash package="bun"
bun add server-only
```

> Next.js implements custom handling of `server-only` and `client-only` import declarations, in order to produce more accurate error for modules used in the wrong environment. The contents of these packages from `NPM` are not used by Next.js.

Author: ```suggestion
Next.js handles `server-only` and `client-only` imports internally to provide clearer error messages when a module is used in the wrong environment. The contents of these packages from NPM are not used by Next.js.
```
